### ISSUE TITLE
[Sonic Generations] 

### ISSUE THREAD

**Issue by sonicgamer1007** [thesupersonic16/HedgeModManager#276]
HMM Info:
```
    Version: 7.5-0
    Args: 
    StartDir: C:\Program Files (x86)\Steam\steamapps\common\Sonic Generations\
    Process Level: User
    Game: Sonic Generations
    SteamGame: Sonic Generations
```

Exception:
```
    Type: UnauthorizedAccessException
    Message: Access to the path 'C:\Program Files (x86)\Steam\steamapps\common\Sonic Generations\d3d9.dll' is denied.
    Source: mscorlib
    Function: Void WinIOError(Int32, System.String)
    StackTrace: 
       at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
       at System.IO.File.InternalDelete(String path, Boolean checkHost)
       at HedgeModManager.MainWindow.UI_OtherLoader_Click(Object sender, RoutedEventArgs e) in C:\Users\SuperSonic16\source\repos\HedgeModManager\HedgeModManager\UI\MainWindow.xaml.cs:line 871
       at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
       at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
       at System.Windows.Controls.Primitives.ButtonBase.OnClick()
       at System.Windows.Controls.Button.OnClick()
       at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
       at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
       at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
       at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
       at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
       at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
       at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
       at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
       at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
       at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
       at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
       at System.Windows.Input.InputManager.ProcessStagingArea()
       at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
       at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
       at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
       at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
       at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
       at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
       at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
       at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
       at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
       at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
       at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
       at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
       at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
       at System.Windows.Application.RunDispatcher(Object ignore)
       at System.Windows.Application.RunInternal(Window window)
       at HedgeModManager.HedgeApp.Main(String[] args) in C:\Users\SuperSonic16\source\repos\HedgeModManager\HedgeModManager\HedgeApp.xaml.cs:line 327
    InnerException: 
```

**Comment by sonicgamer1007** [comment]
my game keeps freezing when modern sonic jumps down.

**Comment by thesupersonic16** [comment]
Crash should be fixed in 65b2f782ebe98eb108aa42c5affd0b9174b30298 which will be released in the next update.

As for your game freezing issue, it could be a mod incompatibility. Does the issue still occur with no mods enabled?

**Comment by sonicgamer1007** [comment]
The game will either stay on sonic falling or stay on the loading screen

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
I even tried turning off all the codes and mods, yesterday it started to not work for no reason



Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by thesupersonic16** [comment]
What Graphics card do you have?

**Comment by sonicgamer1007** [comment]
I use a Mac mini 2012  (higher end) boot amped into windows 10 pro, I have the graphics settings on low and it runs fine

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
The settings I have sonic generations at are 1280x768 60hz, no antialiasing, widescreen, shadow and reflection quality low, and v-sync is on

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
It plays games like bendy and the ink machine at a higher pixel count and more fancy graphical dodads

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by thesupersonic16** [comment]
Can you check if the game works fine without mods? Like just click Uninstall Mod Loader in the settings and press Play.

**Comment by sonicgamer1007** [comment]
The crashing occurs when I try to unistall the loader

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
Let me try again

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
It would be something to do with the custom codes as it says they’re incompatible, and it won’t update them

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by thesupersonic16** [comment]
Make sure the game is closed and try uninstall the loader, After that ignore the codes warning, its there because the modloader is not installed.

**Comment by sonicgamer1007** [comment]
I uninstalled it and it now works, I’m going to try install the codes again and see how it goes

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
Probably it’s trying to load modern sonic instead of classic sonic, but it’s also trying to load classic sonic

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
Let me reset my file

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by sonicgamer1007** [comment]
It works now

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by thesupersonic16** [comment]
Is the issue solved? If so you can close the issue. But if you still have issues with HMM, let us know.

**Comment by sonicgamer1007** [comment]
Once it loads modern sonic it crashes, so I might as well wait 4 years for another computer so it can handle the game better, Thank you for trying to help though.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
